homeassistant/components/kmtronic/switch.py

```diff
@@ -28,22 +30,23 @@ async def async_setup_entry(
 class KMtronicSwitch(CoordinatorEntity, SwitchEntity):
     """KMtronic Switch Entity."""
 
-    def __init__(self, coordinator, relay, reverse, config_entry_id):
+    def __init__(self, hub, coordinator, relay, reverse, config_entry_id):
         """Pass coordinator to CoordinatorEntity."""
         super().__init__(coordinator)
         self._relay = relay
         self._config_entry_id = config_entry_id
         self._reverse = reverse
 
-    @property
-    def name(self) -> str:
-        """Return the name of the entity."""
-        return f"Relay{self._relay.id}"
+        hostname = urllib.parse.urlsplit(hub.host).hostname
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, self._config_entry_id)},
```

Comment:
You don't need to set `self._config_entry_id` anymore.
```suggestion
            "identifiers": {(DOMAIN, config_entry_id)},
```